test/e2e/upgrade_test.go

```diff
@@ -221,11 +222,20 @@ func TestUpgrade(t *testing.T) {
 
 // testWorkloadsAvailable tests that all workloads deployed on Windows nodes by the test suite are available
 func (tc *testContext) testWorkloadsAvailable(t *testing.T) {
-	deployments, err := tc.client.K8s.AppsV1().Deployments(tc.workloadNamespace).List(context.TODO(),
-		metav1.ListOptions{})
-	require.NoError(t, err)
-	for _, deployment := range deployments.Items {
-		assert.Equalf(t, deployment.Spec.Replicas, deployment.Status.AvailableReplicas,
-			"deployment %s is not available", deployment.GetName())
-	}
+	err := wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+		deployments, err := tc.client.K8s.AppsV1().Deployments(tc.workloadNamespace).List(context.TODO(),
+			metav1.ListOptions{})
+		if err != nil {
+			log.Printf("error getting deployment list: %s", err)
+		}
+		for _, deployment := range deployments.Items {
+			if deployment.Spec.Replicas == nil ||
+				(*deployment.Spec.Replicas == deployment.Status.AvailableReplicas) {
```

Comment:
You are adding an extra nil check here. I don't follow how that applies to not updating the current state of the deployments when polling given we were not polling before.